services/powchain/src/main/java/tech/pegasys/teku/services/powchain/Eth1ProviderMonitor.java

```diff
@@ -50,19 +42,14 @@ private void validate() {
     if (stopped.get()) {
       return;
     }
-
-    eth1Provider
-        .getChainId()
-        .whenException(err -> LOG.debug("Failed to get Eth1 chain id. Will retry.", err))
-        .thenAccept(
-            chainId -> {
-              if (chainId.intValueExact() != Constants.DEPOSIT_CHAIN_ID) {
-                STATUS_LOG.eth1DepositChainIdMismatch(
-                    Constants.DEPOSIT_CHAIN_ID, chainId.intValueExact());
-              }
-            })
-        .handleComposed(
-            (__, err) -> asyncRunner.runAfterDelay(this::validate, Duration.ofMinutes(1)))
-        .reportExceptions();
+    eth1ProviderSelector
+        .getProviders()
+        .parallelStream()
+        .forEach(
+            monitorableEth1Provider -> {
+              if (monitorableEth1Provider.needsToBeValidated()) monitorableEth1Provider.validate();
```

Comment:
nit: should include braces for if statements.  Or could just use a `.filter` line.